src/analytics/lambdas/scan-metadata-workflow/store-metadata-into-ddb.ts

```diff
@@ -452,29 +487,173 @@ async function createEventParameterItem(record: any, itemsMap: Map<string, any>,
   };
 }
 
-function aggEventParameterSummary(itemsMap: Map<string, any>) {
+async function aggEventParameterSummary(
+  itemsMap: Map<string, any>,
+  allDistinctEventSet: Set<string>,
+  allDistinctPlatformSet: Set<string>,
+) {
+  const metadataNameSet = getMetadataName();
   for (const item of itemsMap.values()) {
     const platformSet: Set<string> = new Set();
     const valueEnumAggregation: { [key: string]: number } = {};
-    aggEventParameterValueEnumAndPlatform(item, platformSet, valueEnumAggregation);
+    aggEventParameterValueEnumAndPlatform(item, platformSet, valueEnumAggregation, metadataNameSet, allDistinctPlatformSet);
     const valueEnum = [];
     // limit the size of valueEnum to 50
     for (const key in valueEnumAggregation) {
       if (valueEnum.length < 50) {
         valueEnum.push({ value: key, count: valueEnumAggregation[key] });
+      } else {
+        break;
       }
     }
-
     item.summary.platform = Array.from(platformSet);
     item.summary.valueEnum = valueEnum;
+
+    if (metadataNameSet.has(item.name)) {
+      item.summary.associatedEvents = Array.from(allDistinctEventSet);
+      item.summary.platform = Array.from(allDistinctPlatformSet);
+    }
+  }
+}
+
+async function getDistinctEventName(appId: string) {
+  const inputSql = `SELECT distinct event_name FROM ${appId}.event_v2;`;
```

Comment:
I'm not sure why we need it. Do you have a benchmark when running it on a data volume > 10B events?

Reply:
According to requirements,  for build-in event parameters, they should be associated with all event. @luorobin-a2z 
Please confirm this requirements.

Is it possible to have 10B distinct event name?

Reply:
My concern is that it will scan the entire table for performance and cost impact. We need a benchmark for large volumes of data before going ahead.

Reply:
Tested 4B event records and execute "SELECT distinct event_name FROM app1.event_v2;"  in 10 seconds.

Reply:
Tested 4B event records and execute "SELECT distinct event_name FROM app1.event_v2;" in 10 seconds for 16 RPU.